### PR TITLE
fix: improve behavior around cluster connection during deploy

### LINKED ISSUE #2085: Zarf attempting to load Kube context prior to deploy confirm
If an invalid/old kubecontext exists, zarf will try to connect for a very long time and sit in a pending state with no output. Zarf should not try to load context until the first thing that needs to access the cluster.
Screenshot 2023-10-18 at 3 30 33 AM

### PR DESCRIPTION
**Pull request by Noxsios**
## Description

Fixes #2085 

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Other (security config, docs update, etc)

## Checklist before merging

- [ ] Test, docs, adr added or updated as needed
- [x] [Contributor Guide Steps](https://github.com/defenseunicorns/zarf/blob/main/CONTRIBUTING.md#developer-workflow) followed

### CODE REVIEW

File: src/pkg/utils/helpers/misc.go [hunk 1]
```diff
@@ -193,3 +193,12 @@ func StringToSlice(s string) []string {
 
 	return []string{}
 }
+
+func AppendIfNotExists(slice []string, item string) []string {
```
Comment (Noxsios) [inline src/pkg/utils/helpers/misc.go:197]: lint

File: src/test/e2e/29_mismatched_checks_test.go [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+	// Get the current init package secret
+	initPkg := types.DeployedPackage{}
+	base64Pkg, _, err := e2e.Kubectl("get", "secret", "zarf-package-init", "-n", "zarf", "-o", "jsonpath={.data.data}")
+	require.NoError(t, err)
+	jsonPkg, err := base64.StdEncoding.DecodeString(base64Pkg)
+	require.NoError(t, err)
+	fmt.Println(string(jsonPkg))
+	err = json.Unmarshal(jsonPkg, &initPkg)
+	require.NoError(t, err)
```
Comment (Noxsios) [inline src/test/e2e/29_mismatched_checks_test.go:59]: we could replace this w/ `zarf package inspect init -o json` if we wanted to

Comment (Racer159) [reply src/test/e2e/29_mismatched_checks_test.go:59]: we actually can't because this is a DeployedPackage not a ZarfPackage.

Comment (Noxsios) [reply src/test/e2e/29_mismatched_checks_test.go:59]: Then `zarf package list -o json --package=init` could work, no?

**Review by Noxsios — COMMENTED** [review]
lgtm

**Review by Racer159 — APPROVED** [review]
lgtm

**Review by lucasrod16 — APPROVED** [review]
lgtm

**Pull request merged**